pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -238,10 +245,9 @@ func (p *LaunchTemplateProvider) Invalidate(ctx context.Context, ltName string)
 // hydrateCache queries for existing Launch Templates created by Karpenter for the current cluster and adds to the LT cache.
 // Any error during hydration will result in a panic
 func (p *LaunchTemplateProvider) hydrateCache(ctx context.Context) {
-	queryKey := fmt.Sprintf(launchTemplateNameFormat, injection.GetOptions(ctx).ClusterName, "*")
-	p.logger.Debugf("Hydrating the launch template cache with names matching \"%s\"", queryKey)
+	p.logger.Debugf("Hydrating the launch template cache with tags matching \"%s: %s\"", karpenterManagedTagKey, karpenterManagedTagVal)
 	if err := p.ec2api.DescribeLaunchTemplatesPagesWithContext(ctx, &ec2.DescribeLaunchTemplatesInput{
-		Filters: []*ec2.Filter{{Name: aws.String("launch-template-name"), Values: []*string{aws.String(queryKey)}}},
+		Filters: []*ec2.Filter{{Name: aws.String(fmt.Sprintf("tag:%s", karpenterManagedTagKey)), Values: []*string{aws.String(karpenterManagedTagVal)}}},
```

Comment:
Can this cause cross-cluster deletion? Previously we used to use the ClusterName as part of discovery